### PR TITLE
[Improvement]Log be thread num

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
Log doris_be process's thread num at a fixed interval, when a BE cores because of thread exhaustion, we can use this log to find which logic uses the most threads.

Use http to get thread num or find it in be.INFO
```
curl -m 10 http://127.0.0.1:8240/api/be_process_thread_num
```

```
[
    [
        "be_process_id",
        2564391
    ],
    [
        "total_thread_count",
        7104
    ],
    [
        "distinct_thread_name_count",
        106
    ],
    [
        "RScan_auth_grou",
        960
    ],
    [
        "RScan_normal [w",
        960
    ],
    [
        "RScan_test_wg [",
        960
    ],
    [
        "RemoteScanThrea",
        960
    ],
    [
        "brpc_heavy",
        384
    ],
    [
        "brpc_light",
        384
    ],
    [
        "doris_be",
        295
    ],
    [
        "Scan_normal [wo",
        192
    ],
    [
        "Scan_auth_group",
        192
    ],
    [
        "LimitedScanThre",
        192
    ],
    [
        "Scan_test_wg [w",
        192
    ],
    [
        "SpillIOThreadPo",
        192
    ],
    [
        "local_scan",
        192
    ],
    [
        "Pipe_test_wg [w",
        96
    ],
    [
        "Pipe_normal [wo",
        96
    ],
    [
        "Pipe_auth_group",
        96
    ],
    [
        "PipeNoGSchePool",
        96
    ],
    [
        "DownloadThreadP",
        64
    ],
    [
        "SendReportThrea",
        64
    ],
    [
        "SendBatchThread",
        64
    ],
    [
        "FragmentMgrThre",
        64
    ],
    [
        "JoinNodeThreadP",
        64
    ],
    [
        "EvHttpServer [w",
        48
    ],
    [
        "FragmentInstanc",
        48
    ],
    [
        "TabletPublishTx",
        32
    ],
    [
        "S3FileUploadThr",
        16
    ],
    [
        "BufferedReaderP",
        16
    ],
    [
        "NonBlockCloseTh",
        12
    ],
    [
        "MultiGetTaskThr",
        10
    ],
    [
        "TaskWP_PUBLISH_",
        8
    ],
    [
        "SendTableStatsT",
        8
    ],
    [
        "MemTableHighPri",
        6
    ],
    [
        "MemTableFlushTh",
        6
    ],
    [
        "wg_flush_test_w",
        6
    ],
    [
        "wg_flush_auth_g",
        6
    ],
    [
        "SegmentFileWrit",
        6
    ],
    [
        "wg_flush_normal",
        6
    ],
    [
        "WriteCooldownMe",
        5
    ],
    [
        "TaskWP_MAKE_SNA",
        5
    ],
    [
        "CooldownTaskThr",
        5
    ],
    [
        "SegCompactionTa",
        5
    ],
    [
        "TaskWP_RELEASE_",
        5
    ],
    [
        "BaseCompactionT",
        4
    ],
    [
        "cancel_timeout_",
        3
    ],
    [
        "TaskWP_DELETE [",
        3
    ],
    [
        "TaskWP_DROP_TAB",
        3
    ],
    [
        "TaskWP_CREATE_T",
        3
    ],
    [
        "TaskWP_CLONE [w",
        3
    ],
    [
        "TaskWP_ALTER_TA",
        3
    ],
    [
        "TaskWP_ALTER_IN",
        3
    ],
    [
        "TaskWP_.PUSH [w",
        3
    ],
    [
        "HighPriorPool.P",
        3
    ],
    [
        "ColdDataCompact",
        2
    ],
    [
        "memtable_memory",
        1
    ],
    [
        "query_runtime_s",
        1
    ],
    [
        "memory_maintena",
        1
    ],
    [
        "memory_gc_threa",
        1
    ],
    [
        "GroupCommitThre",
        1
    ],
    [
        "je_purge_dirty_",
        1
    ],
    [
        "gc_expired_cont",
        1
    ],
    [
        "garbage_sweeper",
        1
    ],
    [
        "evict_querying_",
        1
    ],
    [
        "disk_stat_monit",
        1
    ],
    [
        "cooldown_tasks_",
        1
    ],
    [
        "compaction_task",
        1
    ],
    [
        "cold_data_compa",
        1
    ],
    [
        "path_gc_thread-",
        1
    ],
    [
        "ping_worker-256",
        1
    ],
    [
        "clean_expired_t",
        1
    ],
    [
        "remove_unused_r",
        1
    ],
    [
        "replay_wal-2566",
        1
    ],
    [
        "spill_gc_thread",
        1
    ],
    [
        "tablet_checkpoi",
        1
    ],
    [
        "unused_rowset_m",
        1
    ],
    [
        "update_wal_dir_",
        1
    ],
    [
        "GroupCommitRepl",
        1
    ],
    [
        "CumuCompactionT",
        1
    ],
    [
        "wg_mem_refresh_",
        1
    ],
    [
        "workload_schedu",
        1
    ],
    [
        "TaskWP_UPDATE_T",
        1
    ],
    [
        "TaskWP_CLEAN_UD",
        1
    ],
    [
        "TaskWP_CLEAR_TR",
        1
    ],
    [
        "TaskWP_CHECK_CO",
        1
    ],
    [
        "TaskWP_DOWNLOAD",
        1
    ],
    [
        "TaskWP_GC_BINLO",
        1
    ],
    [
        "TabletCalcDelet",
        1
    ],
    [
        "TaskWP_MOVE [wo",
        1
    ],
    [
        "SingleReplicaCo",
        1
    ],
    [
        "TaskWP_PUSH_COO",
        1
    ],
    [
        "TaskWP_PUSH_STO",
        1
    ],
    [
        "REPORT_TASK-256",
        1
    ],
    [
        "TaskWP_STORAGE_",
        1
    ],
    [
        "TaskWP_SUBMIT_T",
        1
    ],
    [
        "clean_idle_cons",
        1
    ],
    [
        "TaskWP_UPDATE_V",
        1
    ],
    [
        "TaskWP_UPLOAD [",
        1
    ],
    [
        "REPORT_OLAP_TAB",
        1
    ],
    [
        "_update_replica",
        1
    ],
    [
        "async_publish_v",
        1
    ],
    [
        "be_proc_monitor",
        1
    ],
    [
        "REPORT_DISK_STA",
        1
    ],
    [
        "MigrationTaskTh",
        1
    ],
    [
        "cache_clean_thr",
        1
    ],
    [
        "calculate_metri",
        1
    ],
    [
        "LazyReleaseMemo",
        1
    ],
    [
        "TaskWP_CLEAN_TR",
        1
    ]
]
```

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

File: be/src/runtime/be_proc_monitor.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+#include <filesystem>
+#include <fstream>
+#include <map>
+
+std::string BeProcMonitor::get_be_thread_info() {
+    int32_t pid = getpid();
+    std::string proc_path = fmt::format("/proc/{}/task", pid);
+    if (access(proc_path.c_str(), F_OK) != 0) {
+        LOG(INFO) << "be proc path " << proc_path << " not exists.";
```
Comment (HappenLee) [inline be/src/runtime/be_proc_monitor.cpp:33]: better warning log

File: be/src/common/daemon.cpp [hunk 2]
```diff
@@ -396,6 +397,14 @@ void Daemon::wg_mem_used_refresh_thread() {
     }
 }
 
+void Daemon::be_proc_monitor_thread() {
+    while (!_stop_background_threads_latch.wait_for(
+            std::chrono::milliseconds(config::be_proc_monitor_interval_ms))) {
+        BeProcMonitor bpm;
```
Comment (HappenLee) [inline be/src/common/daemon.cpp:403]: should we create a object each while loop，better the `get_be_thread_info()` should be static

File: be/src/runtime/be_proc_monitor.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+#include <fmt/format.h>
+#include <glog/logging.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <filesystem>
+#include <fstream>
+#include <map>
+
+std::string BeProcMonitor::get_be_thread_info() {
```
Comment (HappenLee) [inline be/src/runtime/be_proc_monitor.cpp:29]: why direct log in the function ?

File: be/src/runtime/be_proc_monitor.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                continue;
+            }
+            std::stringstream str_buf;
+            str_buf << file.rdbuf();
+            std::string thread_name = str_buf.str();
+            thread_name.erase(std::remove(thread_name.begin(), thread_name.end(), '\n'),
+                              thread_name.end());
+
+            if (thread_num_map.find(thread_name) != thread_num_map.end()) {
+                int count = thread_num_map.at(thread_name);
```
Comment (HappenLee) [inline be/src/runtime/be_proc_monitor.cpp:58]: has do the find, no need to do `at`-> `[]`

File: be/src/runtime/be_proc_monitor.cpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                count++;
+                thread_num_map[thread_name] = count;
```
Comment (HappenLee) [inline be/src/runtime/be_proc_monitor.cpp:60]: thread_num_map[thread_name]++;

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**